libs/model/src/featurelayer.cpp

```diff
@@ -0,0 +1,248 @@
+#include "featurelayer.h"
+
+#include <unordered_map>
+
+namespace mapget
+{
+
+struct TileFeatureLayer::Impl {
+    std::optional<model_ptr<Object>> featureIdPrefix_;
+
+    sfl::segmented_vector<Feature::Data, simfil::detail::ColumnPageSize/4> features_;
+    sfl::segmented_vector<Attribute::Data, simfil::detail::ColumnPageSize> attributes_;
+    sfl::segmented_vector<FeatureId::Data, simfil::detail::ColumnPageSize/2> featureIds_;
+    sfl::segmented_vector<simfil::ArrayIndex, simfil::detail::ColumnPageSize/2> attrLayers_;
+    sfl::segmented_vector<simfil::ArrayIndex, simfil::detail::ColumnPageSize/2> attrLayerLists_;
+
+    // Simfil execution Environment for this tile's string pool.
+    simfil::Environment simfilEnv_;
+
+    // Compiled simfil expressions, by hash of expression string
+    std::unordered_map<size_t, simfil::ExprPtr> simfilExpressions_;
+
+    Impl(std::shared_ptr<Fields> fields)
+        : simfilEnv_(std::move(fields))
+    {
+    }
+};
+
+TileFeatureLayer::TileFeatureLayer(
+    TileId tileId,
+    std::string const& nodeId,
+    std::string const& mapId,
+    std::shared_ptr<LayerInfo> const& layerInfo,
+    std::shared_ptr<Fields> const& fields
+) :
+    simfil::ModelPool(fields),
+    impl_(std::make_unique<Impl>(fields)),
+    TileLayer(tileId, nodeId, mapId, layerInfo)
+{
+}
+
+TileFeatureLayer::~TileFeatureLayer() {}  // NOLINT
```

Comment:
What is this `NOLINT` surpressing?
Use `= default` instead of `{}` here.

Reply:
`Use = default instead of {} here.`

Haha, that is why NOLINT is there, because clang-tidy also wants that. But we can't, because of the pimpl. I'll add a comment.